### PR TITLE
Add dotnet path for Ubuntu 19.04 for AssemblyResolvers

### PR DESCRIPTION
**Pull request by joelhoisko**
Tested on Ubuntu 19.04 on a dolittle project.

Fixes issues https://github.com/dolittle-fundamentals/DotNET.Fundamentals/issues/258  and https://github.com/dolittle-fundamentals/DotNET.Fundamentals/issues/244, not the prettiest fix and in the future we probably could do with letting the user specify the path or having a more robust/general dotnet path discovery function/tool.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1115616103797671/1146246334480284)

### CODE REVIEW

File: Source/Assemblies/NuGetFallbackFolderAssemblyResolver.cs [hunk 1]
```diff
@@ -14,17 +14,30 @@ namespace Dolittle.Assemblies
...
+            string basePath;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                basePath = @"c:\Program Files\dotnet\sdk\NuGetFallbackFolder";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                // default location on Ubuntu 19.04
+                basePath = "/usr/share/dotnet/sdk/NuGetFallbackFolder";
+            }
```
Comment (TomasEkeli) [inline Source/Assemblies/NuGetFallbackFolderAssemblyResolver.cs]: How will this behave on machines with Linux but not on Ubuntu 19.04 or newer? That's a small subset of Linuxen.

**Comment by TomasEkeli** [comment]
Should we have multiple fallback paths, and try each and every one if we fall back? That way we can just add new ones when we discover a platform with this problem. 

Or, you know, everyone could use the same environment ;)

File: Source/Assemblies/NuGetFallbackFolderAssemblyResolver.cs (hunk 1 above)
Comment (joelhoisko) [reply Source/Assemblies/NuGetFallbackFolderAssemblyResolver.cs]: It should be the same location for Ubuntu 18.04 at least too and doing a crude search for the location from the official [dotnet-setup repo](https://github.com/dotnet/core-setup/search?p=1&q=%2Fusr%2Fshare&unscoped_q=%2Fusr%2Fshare) at least seemingly shows that they use `/usr/share` for rpm and deb based installations (or that's my review of the search at a glance).

**Comment by joelhoisko** [comment]
Having a specific tool/function to find the correct path from a list of possibilities would be great and then if none of those paths work we could give the user the chance to specify their own path, aka `--dotnet-path=/users/weird/dotnet/folder/` or/and have a configuration file for this. This would require us to fail earlier with a more clear error message too instead of complaining about some random missing dll's basically.

File: Source/Assemblies/NuGetFallbackFolderAssemblyResolver.cs (hunk 1 above)
Comment (TomasEkeli) [reply Source/Assemblies/NuGetFallbackFolderAssemblyResolver.cs]: In that case - why has this not been failing consistently for all of us when building on linux?

I am not knowledgeable enough about the Linux ecosystem to review or accept this PR. I summon @einari

Comment (einari) [reply Source/Assemblies/NuGetFallbackFolderAssemblyResolver.cs]: This is used at Runtime - and our build tool uses this and could potentially have the problem. But it has very limited number of references, just the BCL and Dolittle + Autofac for the most part, while where these folders matter is when you have things that are part of what is considered the "Global Assembly Cache" - (yeah you heard it here, I pointed at the elephant😁). For inshance ASP.net stuff is typically found in one of these.

**Comment by einari** [comment]
These paths shouldn't be that dynamic and you should be able to get them by doing "dotnet info" - which suggests we should tap into what they're doing themselves.

**Pull request merged**